### ISSUE TITLE
Gutenberg does not display custom taxonomies

### ISSUE THREAD

**Issue by thebrowser** [WordPress/gutenberg#19289]
Hello, I've encountered an issue while creating custom taxonomies and then trying to add them to a post/page/custom post while using the Gutenberg editor. The issue cannot be reproduced on the classic editor.

The issue is while using Gutenberg, on the right side menu the child taxonomies don’t appear and I also cannot add any new ones from there. Note that the custom slug that I selected does show up, it’s just not functional.

I’ve read online some posts about how displaying the custom taxonomies is an issue sometimes, but ny issue is specifically about selecting/adding/deleting new child taxonomies while working on a post. Also, "hierarchical" => false seems to work in the sense that I’m able to add new tags. However these are not saved; when I save, leave and reopen that blog post in the admin dashboard the tags are all gone.

This is my code: 

```
function my_custom_taxonomies(){
  $labels = array(
    'name'              => 'Sections',
    'singular_name'     => 'Section',
    'search_items'      => 'All Sections',
    'all_items'         => 'All Sections',
    'parent_item'       => 'Parent Section',
    'parent_item_colon' => 'Parent Section:',
    'edit_item'         => 'Edit Section',
    'update_item'       => 'Update Section',
    'add_new_item'      => 'Add New Section',
    'new_item_name'     => 'New Section Name',
    'menu_name'         => 'Sections'
  );

  $args = array(
    'labels'            => $labels,
    'hierarchical'      => true,
    'public'            => true,
    'show_ui'           => true,
    'show_in_rest'      => true,
    'show_in_menu'      => true,
    'show_in_nav_menus' => true,
    'show_admin_column' => true,
    'query_var'         => true,
    'rewrite'           => array('slug' => 'section')
  );

  register_taxonomy('sections', array('post', 'page'), $args);
}

add_action('init', 'my_custom_taxonomies');
```

For now the obvious workaround is to use the classic editor which I'm happy to use, but I'd be nicer if Gutenberg did support this as well.

From the screenshots attached you can see that I have added a few taxonomies, however from Gutenberg the "Section" (name of the taxonomy) appears but doesnt actually work in any way beyond that.

Thanks,

![taxonomies2](https://user-images.githubusercontent.com/36519816/71308407-205a4280-23f4-11ea-82ae-75b252353d1c.png)
![taxonomies1](https://user-images.githubusercontent.com/36519816/71308408-205a4280-23f4-11ea-83ea-1d3b64f2764c.png)

**Comment by kevinfodness** [comment]
It seems like the REST request to save the taxonomy is failing. Can you look in the Network tab of the developer tools, spot the request to save, and look at what error message it gives you? Typically, the error messages are very specific and will let you know what went wrong (e.g., you don't have permission to add terms to this taxonomy).

**Comment by thebrowser** [comment]
Hi, thank you for your help!

I found two _400 Bad Request_ errors in the network tab that trigger at different times. The first one is a GET request when I click on the arrow to reveal the taxonomies (I named them sections) and the second one is POST, when I click on the _Add New Section_ button.

I've read the messages but they don't make much sense in this context. I have only included the code from my previous message. Here's a screenshot that I took from both errors as I triggered them:

![rest-error](https://user-images.githubusercontent.com/36519816/71401948-5b8e8880-2623-11ea-99c1-917aa308f9d7.png)
